packages/react/direction/src/Direction.tsx

```diff
@@ -0,0 +1,33 @@
+import * as React from 'react';
+
+type Direction = 'ltr' | 'rtl';
+const DirectionContext = React.createContext<Direction | undefined>(undefined);
+
+/* -------------------------------------------------------------------------------------------------
+ * Direction
+ * -----------------------------------------------------------------------------------------------*/
+
+interface DirectionProviderProps {
+  dir: Direction;
+}
+const DirectionProvider: React.FC<DirectionProviderProps> = (props) => {
+  const { dir, children } = props;
+  return <DirectionContext.Provider value={dir}>{children}</DirectionContext.Provider>;
+};
+
+/* -----------------------------------------------------------------------------------------------*/
+
+function useDirection(directionProp?: Direction) {
```

Comment:
Some initial feedback:

Not sure if I'm missing the point here, but it seems strange to me to pass in the dir prop inside the hook, just to use that prop if defined instead of the context, rather than do that `||` on the usage side.

What's the advantage?

We're usually checking the `dir` prop in one place at the top anyway and then exposing that through context, so really the fallback check would happen only in one place anyway. Seems to make more sense that way.

I'm probably missing something obvious… 😀